core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -984,6 +979,29 @@ public NativeImageInvokerInfo build() {
                 if (nativeConfig.monitoring().isPresent()) {
                     monitoringOptions.addAll(nativeConfig.monitoring().get());
                 }
+
+                if (!monitoringOptions.isEmpty() && monitoringOptions.contains(NativeConfig.MonitoringOption.NONE)) {
```

Comment:
Hmmm, this double iff is a bit confusing to read with the duplication of the condition. What about this?

```
if (monitoringOptions.remove(NativeConfig.MonitoringOption.NONE) 
    && !monitoringOptions.isEmpty()) {
  // log warning
}
```

Reply:
Nice. Sure!